StRoot/StFstHitMaker/StFstHitMaker.cxx

```diff
@@ -0,0 +1,214 @@
+		return kStWarn;
+	}
+
+	// Get pointer to an existing StFstHitCollection if any
+	StFstHitCollection *fstHitCollection = eventPtr->fstHitCollection();
+
+	// If no fst hit collection, create one
+	if (!fstHitCollection) {
+		fstHitCollection = new StFstHitCollection();
+		eventPtr->setFstHitCollection(fstHitCollection);
+		LOG_DEBUG << "Make() - Added new StFstHitCollection to this StEvent" << endm;
+	}
+
+	unsigned char  nClusteringType = -1;
+
+	for (unsigned char wedgeIdx = 0; wedgeIdx < kFstNumWedges; ++wedgeIdx) {
+		//add new hits from clusters
+
+		StFstClusterCollection *clusterCollectionPtr = fstCollectionPtr->getClusterCollection(wedgeIdx );
+
+		if ( clusterCollectionPtr ) {
+			unsigned int numClusters = clusterCollectionPtr->getNumClusters();
+			LOG_DEBUG << "Make() - Number of clusters found in wedge " << (int)(wedgeIdx + 1) << ": " << numClusters << endm;
+
+			unsigned short idTruth = 0;
+			unsigned char  nRawHits = -1, nRawHitsR = -1, nRawHitsPhi = -1;
+			unsigned char  disk = -1, wedge = -1, sensor = -1, apv = -1;
+			int  meanRStrip = -1, meanPhiStrip = -1;
+                        float  charge = 0., chargeErr = 0.;
+			unsigned char  maxTb = -1;
+			int	 key = -1;
+			float phiInner = -999.9, phiOuter = -999.9;
+
+			for (std::vector< StFstCluster * >::iterator clusterIter = clusterCollectionPtr->getClusterVec().begin(); clusterIter != clusterCollectionPtr->getClusterVec().end(); ++clusterIter) {
+				idTruth         = (*clusterIter)->getIdTruth();
+				key             = (*clusterIter)->getKey();
+				disk            = (*clusterIter)->getDisk();
+				wedge           = (*clusterIter)->getWedge();
+				sensor          = (*clusterIter)->getSensor();
+				apv             = (*clusterIter)->getApv();
+				meanRStrip      = (*clusterIter)->getMeanRStrip();
+				meanPhiStrip    = (*clusterIter)->getMeanPhiStrip();
+				maxTb           = (*clusterIter)->getMaxTimeBin();
+				charge          = (*clusterIter)->getTotCharge();
+				chargeErr       = (*clusterIter)->getTotChargeErr();
+				nRawHits        = (*clusterIter)->getNRawHits();
+				nRawHitsR       = (*clusterIter)->getNRawHitsR();
+				nRawHitsPhi     = (*clusterIter)->getNRawHitsPhi();
+				nClusteringType = (*clusterIter)->getClusteringType();
+
+				StFstHit *newHit = new StFstHit(disk, wedge, sensor, apv, charge, chargeErr, maxTb, meanRStrip, meanPhiStrip, nRawHits, nRawHitsR, nRawHitsPhi);
+				newHit->setId(key);
+				newHit->setIdTruth(idTruth);
+
+                                int moduleIdx;
+				if(disk == 1) // Disk 1
+					moduleIdx = wedge;
+				if(disk == 2)// Disk 2
+					moduleIdx = wedge-12;
+				if(disk == 3)// Disk 3
+					moduleIdx = wedge-24;
+				if(disk == 1 || disk == 3)
+				{// Disk 1 & 3
+					phiInner = kFstphiStart[moduleIdx-1]*TMath::Pi()/6.0 + 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+					phiOuter = kFstphiStop[moduleIdx-1]*TMath::Pi()/6.0  - 0.5*kFstzDirct[moduleIdx-1]*kFstStripPitchPhi;
+				}
+				if(disk == 2)
```

Comment:
should this not rather be "else if (disk == 2) "
.No reason to test if condition already has bee full-filled .
This appears many times in the code; I do not think, but could be wrong that compiler can optimize this?

Reply:
I updated these long if chains to use if(....) else if(....) else wherever appropriate. Done in 55239713e11e09ccab2c8d6692c186aa80294da0